boards/nucleo-f401/include/periph_conf.h

```diff
@@ -0,0 +1,390 @@
+#define I2C_0_SCL_CLKEN()   (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOBEN)
+#define I2C_0_SDA_PORT      GPIOB
+#define I2C_0_SDA_PIN       9
+#define I2C_0_SDA_AF        4
+#define I2C_0_SDA_CLKEN()   (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOBEN)
+
+/* I2C 1 device configuration */
+#define I2C_1_DEV           I2C3
+#define I2C_1_CLKEN()       (RCC->APB1ENR |= RCC_APB1ENR_I2C3EN)
+#define I2C_1_CLKDIS()      (RCC->APB1ENR &= ~(RCC_APB1ENR_I2C3EN))
+#define I2C_1_EVT_IRQ       I2C3_EV_IRQn
+#define I2C_1_EVT_ISR       isr_i2c3_ev
+#define I2C_1_ERR_IRQ       I2C3_ER_IRQn
+#define I2C_1_ERR_ISR       isr_i2c3_er
+/* I2C 1 pin configuration */
+#define I2C_1_SCL_PORT      GPIOA
+#define I2C_1_SCL_PIN       8
+#define I2C_1_SCL_AF        4
+#define I2C_1_SCL_CLKEN()   (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOAEN)
+#define I2C_1_SDA_PORT      GPIOB
+#define I2C_1_SDA_PIN       4
+#define I2C_1_SDA_AF        9
+#define I2C_1_SDA_CLKEN()   (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOBEN)
+/** @} */
+
+/**
+ * @name GPIO configuration
+ * @{
+ */
+#define GPIO_NUMOF          3
+#define GPIO_0_EN           1
+#define GPIO_1_EN           1
+#define GPIO_2_EN           1
+#define GPIO_IRQ_PRIO       1
+
+/* IRQ config */
+#define GPIO_IRQ_0          -1/* not configured */
+#define GPIO_IRQ_1          -1/* not configured */
+#define GPIO_IRQ_2          -1/* not configured */
+#define GPIO_IRQ_3          -1/* not configured */
+#define GPIO_IRQ_4          -1/* not configured */
+#define GPIO_IRQ_5          -1/* not configured */
+#define GPIO_IRQ_6          -1/* not configured */
+#define GPIO_IRQ_7          -1/* not configured */
+#define GPIO_IRQ_8          -1/* not configured */
+#define GPIO_IRQ_9          -1/* not configured */
+#define GPIO_IRQ_10         -1/* not configured */
+#define GPIO_IRQ_11         -1/* not configured */
+#define GPIO_IRQ_12         -1/* not configured */
+#define GPIO_IRQ_13         GPIO_0
+#define GPIO_IRQ_14         -1/* not configured */
+#define GPIO_IRQ_15         -1/* not configured */
+
+/* GPIO channel 0 config */
+#define GPIO_0_PORT         GPIOC                   /* Used for user button 1 */
+#define GPIO_0_PIN          13
+#define GPIO_0_CLK          2                       /* 0: PORT A, 1: B ... */
+#define GPIO_0_EXTI_CFG()   (SYSCFG->EXTICR[3] |= SYSCFG_EXTICR4_EXTI13_PC)
+#define GPIO_0_IRQ          EXTI15_10_IRQn
+
+/* GPIO channel 1 config */
+#define GPIO_1_PORT         GPIOC                   /* Used for user button 1 */
+#define GPIO_1_PIN          2
+#define GPIO_1_CLK          2                       /* 0: PORT A, 1: B ... */
+#define GPIO_1_EXTI_CFG()   (SYSCFG->EXTICR[0] |= SYSCFG_EXTICR1_EXTI2_PC)
+#define GPIO_1_IRQ          EXTI2_IRQn
+
+/* GPIO channel 2 config */
+#define GPIO_2_PORT         GPIOC                   /* Used for user button 1 */
+#define GPIO_2_PIN          3
+#define GPIO_2_CLK          2                       /* 0: PORT A, 1: B ... */
+#define GPIO_2_EXTI_CFG()   (SYSCFG->EXTICR[0] |= SYSCFG_EXTICR1_EXTI3_PC)
```

Comment:
in https://github.com/RIOT-OS/RIOT/pull/3095 we changed the implementation and gpio interface. please adapt to that new style, for an example you can check out the stm32f4-discovery board